enigma-tools-m/src/common/utils.rs

```diff
@@ -2,6 +2,21 @@ use crate::localstd::string::String;
 use enigma_crypto::hash::Keccak256;
 use rustc_hex::ToHex;
 
+#[cfg(feature = "sgx")]
+use crate::localstd::sync::{SgxMutex as Mutex, SgxMutexGuard as MutexGuard};
+
+#[cfg(feature = "std")]
+use crate::localstd::sync::{Mutex, MutexGuard};
+
+
+pub trait LockExpectMutex<T> {
+    fn lock_expect(&self, name: &str) -> MutexGuard<T>;
+}
+
+impl<T> LockExpectMutex<T> for Mutex<T> {
+    fn lock_expect(&self, name: &str) -> MutexGuard<T> { self.lock().unwrap_or_else(|_| panic!("{} mutex is poison", name)) }
```

Comment:
I know that this comes from the original code and is not related directly to this refactoring, but it is important enough. Do we want `panic` here and not SystemError?

Reply:
that's a good question.
generally as long as we don't have threads this will never panic.

this panics only if the mutex is poisoned, and the only way it can get poisoned is by a thread holding the lock panics without releasing it.

btw, you usually do want to panic here to propagate the panic between the threads.

https://doc.rust-lang.org/std/sync/struct.Mutex.html#poisoning

Reply:
Ok, I see. It proves again how our code even conceptually is not ready for the multi-threading.